File: C++/int_array_mem.cpp
#include <chrono>
#include <iostream>
#include <unistd.h>

using namespace std;

int main() {
    const int ONE_BILLION = 1000000000;
    const int ONE_MILLION = 1000000;
    int x = 0;
    static int ints[ONE_MILLION];
    while (x < ONE_MILLION) {
        ints[x] = x;
        x++;
    }
    string temp;
    cin >> temp;
}

File: Java/IntArrayMem.java
import java.util.Scanner;

class IntArrayMem {
    public static void main(String[] args) {
        final int ONE_BILLION = 1000000000;
        final int ONE_MILLION = 1000000;
           
        int x = 0;
        int[] ints = new int[ONE_MILLION];
        while (x < ONE_MILLION) {
            ints[x] = x;
            x++;
        }

        System.gc();

        Scanner scanner = new Scanner(System.in);
        scanner.nextLine();
        scanner.close();

    }
}